includes/data-stores/class-wc-admin-reports-coupons-data-store.php

```diff
@@ -58,6 +58,7 @@ public static function init() {
 		add_action( 'save_post', array( __CLASS__, 'sync_order_coupons' ) );
 		add_action( 'clean_post_cache', array( __CLASS__, 'sync_order_coupons' ) );
 		add_action( 'woocommerce_order_refunded', array( __CLASS__, 'sync_order_coupons' ) );
+		add_action( 'woocommerce_reports_orders_stats_data_deleted', array( __CLASS__, 'sync_on_order_delete' ), 10 );
```

Comment:
One thing I noticed is that this is only fired when an order is deleted permanently.  I'm wondering if we should also fire this hook when an order is trashed as users might be confused about the resulting data.

Or should we keep the behavior as is and make a note of this in user docs?

Reply:
We can exclude when an order is moved to trash, also we can restore it if the order is restored. No problem.